WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,166 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using System.Threading;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+
+		public WebcamQrReader(Network network)
+		{
+			Network = network;
+		}
+
+		public void StartScanning()
+		{
+			if (ScanningTask is not { })
+			{
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
+					try
+					{
+						if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+						{
+							throw new NotImplementedException("This operating system is not supported.");
+						}
```

Comment:
Why to start scanning just to find out that it is not supported?

Reply:
True, unnecessary